hedera-node/docker/Dockerfile

```diff
@@ -1,25 +1,90 @@
-## Installs OpenJDK17 and openssl (used by Swirlds Platform to
+## Installs OpenJDK21 and openssl (used by Swirlds Platform to
 ## generate node keys for e.g. signing states), then copies
 ## required libraries and startup assets for a node with:
 ##  * Configuration from /opt/hedera/services/config-mount; and,
 ##  * Logs at /opt/hedera/services/output; and,
 ##  * Saved states under /opt/hedera/services/output
 ## Ideally we'd use a much lighter (perhaps CoreOS) base, but this will do for now.
+FROM ubuntu:22.04 AS openjdk-builder
+
+# Define Standard Environment Variables
+ENV LC_ALL=C.UTF-8
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install basic OS utilities for building
+RUN apt-get update && \
+	apt-get install --yes tar gzip gnupg2 curl && \
+    apt-get autoclean && \
+    apt-get clean all && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /var/cache/apt
+
+##########################
+####    Java Setup    ####
+##########################
+RUN set -eux; \
```

Comment:
Why are we installing via tarball when the O/S repositories have a fully tested package
`openjdk-21-jdk` we could install with `apt`?

Reply:
Note, the Headless version is also available.